prototype2/gdgem/gdgem.cpp

```diff
@@ -221,23 +220,49 @@ void NMX::input_thread() {
   }
 }
 
+void bin(Hists& hists, const Event &e)
+{
+  uint32_t sum = e.x.adc_sum + e.y.adc_sum;
+  hists.bincluster(sum);
+}
+
+void bin(Hists& hists, const Hit &e)
+{
+  if (e.plane_id == 0) {
+    hists.binstrips(e.strip, e.adc, 0, 0);
+  } else if (e.plane_id == 1) {
+    hists.binstrips(0, 0, e.strip, e.adc);
+  }
+}
+
+void bin_hists(Hists& hists, const std::list<Cluster>& cl)
+{
+  for (const auto& cluster : cl)
+    for (const auto& e : cluster.entries)
+      bin(hists, e);
+}
+
+
 void NMX::processing_thread() {
-  XTRACE(PROCESS, ALW, "NMX Config file: %s", NMXSettings.ConfigFile.c_str());
-  nmx_opts = NMXConfig(NMXSettings.ConfigFile);
   init_builder();
   if (!builder_) {
     XTRACE(PROCESS, ERR, "No builder specified, exiting thread");
     return;
   }
 
   Producer eventprod(EFUSettings.KafkaBroker, "NMX_detector");
-  FBSerializer flatbuffer(kafka_buffer_size, eventprod);
+  EV42Serializer flatbuffer(kafka_buffer_size, "nmx");
+  flatbuffer.producerCallback(
+      std::bind(&Producer::produce2<uint8_t>, &eventprod, std::placeholders::_1));
 
   Producer monitorprod(EFUSettings.KafkaBroker, "NMX_monitor");
   TrackSerializer trackfb(256, nmx_opts.track_sample_minhits,
                           nmx_opts.time_config.target_resolution_ns());
-  HistSerializer histfb;
-  NMXHists hists;
+  Hists hists(Hit::strip_max_val, Hit::adc_max_val);
+  HistSerializer histfb(hists.needed_buffer_size());
```

Comment:
LLVM:ify class.

Reply:
Same goes for NMX stuff. Doro might be working on these classes already, so I'd rather not touch more than is strictly necessary.